ds3-autogen-c/src/main/resources/templates/header-templates/ds3_h.ftl

```diff
@@ -143,25 +162,35 @@ LIBRARY_API void        ds3_client_register_net(ds3_client* client, ds3_error* (
                                                                                                 size_t (*write_handler_func)(void*, size_t, size_t, void*),
                                                                                                 ds3_string_multimap** return_headers));
 LIBRARY_API void ds3_client_proxy(ds3_client* client, const char* proxy);
+LIBRARY_API void ds3_creds_free(ds3_creds* client);
+LIBRARY_API void ds3_client_free(ds3_client* client);
 
+LIBRARY_API ds3_error* ds3_get_object_with_metadata(const ds3_client* client,
+                                                    const ds3_request* request,
+                                                    void* user_data,
+                                                    size_t (* callback)(void*, size_t, size_t, void*),
+                                                    ds3_metadata** _metadata);
 
 <#-- **************************************** -->
 <#-- Generate all "RequestPrototypes"         -->
 <#list getRequests() as requestEntry>
 
     <#if requestEntry.getName() == "head_bucket_request">
+LIBRARY_API ds3_request* ds3_init_head_bucket(const char* bucket_name);
 LIBRARY_API ds3_error* head_bucket_request(const ds3_client* client, const ds3_request* request);
     <#elseif requestEntry.getName() == "head_object_request">
+LIBRARY_API ds3_request* ds3_init_head_object(const char* bucket_name, const char* object_name);
 LIBRARY_API ds3_error* head_object_request(const ds3_client* client, const ds3_request* request, ds3_metadata** _metadata);
     <#else>
+        <#include "InitRequestPrototype.ftl">
         <#include "RequestPrototype.ftl">
     </#if>
 </#list>
 <#-- **************************************** -->
 
 
 LIBRARY_API void ds3_cleanup(void);
-LIBRARY_API void ds3_print_request(const ds3_request* request);
+//LIBRARY_API void ds3_print_request(const ds3_request* request); // No implementation found
```

Comment:
This function, as far as I can tell, is not actually implemented in the current C SDK